image_geometry/src/pinhole_camera_model.cpp

```diff
@@ -56,11 +72,11 @@ bool update(const T& new_val, T& my_val)
   return true;
 }
 
-// For boost::array, std::vector
+// For std::array, std::vector
 template<typename MatT>
 bool updateMat(const MatT& new_mat, MatT& my_mat, cv::Mat_<double>& cv_mat, int rows, int cols)
 {
-  if ((my_mat == new_mat) && (my_mat.size() == cv_mat.rows*cv_mat.cols))
+  if ((my_mat == new_mat) && (my_mat.size() == (unsigned)(cv_mat.rows*cv_mat.cols)))
```

Comment:
Normally we do C++ style casts. If we're aiming to put this in `ros2.repos` then I'd ask you to investigate if this would be better done with a C++ style cast. I'm not quite sure what the right C++ alternative would be, I guess a `reinterpret_cast`, but I don't know if you can just put `unsigned` when you do that. Basically I think you'd need to know the size of the multiplication result too, and include that with the `unsigned`. Maybe there's no good C++ way to do this.